### PR TITLE
[10.x] Allow creation of PSR request with merged data

### LINKED ISSUE #48626: symfony/psr-http-message-bridge v2.3.1 loses the transformations of ConvertEmptyStringsToNull middleware
### Laravel Version

10.26.2

### PHP Version

8.1.24

### Database Driver & Version

_No response_

### Description

Recreating JSON request bodies was changed in v2.3.1 of `symfony/psr-http-message-bridge`.

https://github.com/symfony/psr-http-message-bridge/commit/ef03b6dfb040da9615a11679306d707da7fee51f

The JSON request body is read using `$symfonyRequest->getContent()` instead of `$symfonyRequest->getPayload()->all()` in v2.3.0, and `$symfonyRequest->request->all()` in v2.2.0.

Because of this, the modifications made to a Laravel request object by the `ConvertEmptyStringsToNull` middleware gets lost, i.e. empty string values for JSON request keys are not converted to `null`, and remain as empty strings.

This only affects controller methods that have `$request` type hinted as `Psr\Http\Message\ServerRequestInterface`.

### Steps To Reproduce

I have created a test in a new repository to demonstrate the issue:

```shell
git clone https://github.com/HealthengineAU/laravel-bug-report
```

Tests should fail:

```shell
php artisan test
```

If we install the previous version:

```shell
composer require symfony/psr-http-message-bridge=2.3.0
```

We should see the tests pass:

```
php artisan test
```

### PR DESCRIPTION
**Pull request by timacdonald**
There was a breaking change, for us, in `symfony/psr-http-message-bridge` (https://github.com/symfony/psr-http-message-bridge/pull/122) that means merged data for JSON requests is no longer added to the PSR request. 

The original PR was a fix for a different Laravel issue (although I'm not entirely sure what as the issue tracker for that repo is no longer visible).

fixes #48626

### CODE REVIEW

**Comment by timacdonald** [comment]
uh! dependency troubles. drafting till I fix.

**Comment by mbabker** [comment]
> The original PR was a fix for a different Laravel issue (although I'm not entirely sure what as the issue tracker for that repo is no longer visible).

That was #47834 and basically `InputBag` related issue since.

File: composer.json [hunk 1]
```diff
@@ -44,7 +44,7 @@
         "symfony/console": "^6.2",
         "symfony/error-handler": "^6.2",
         "symfony/finder": "^6.2",
-        "symfony/http-foundation": "^6.2",
+        "symfony/http-foundation": "^6.3",
```
Comment (timacdonald) [inline composer.json:47]: `Request::getPayload` method is only available from `6.3`.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -82,7 +82,9 @@ jobs:
         with:
           timeout_minutes: 5
           max_attempts: 5
-          command: composer update --${{ matrix.stability }} --prefer-dist --no-interaction --no-progress
+          command: |
+            composer require symfony/psr-http-message-bridge:^2.0 nyholm/psr7:^1.0 --no-interaction --no-update
+            composer update --${{ matrix.stability }} --prefer-dist --no-interaction --no-progress
```
Comment (timacdonald) [inline .github/workflows/tests.yml:87]: The optional dependencies are required to run these tests.

Comment (driesvints) [reply .github/workflows/tests.yml:87]: @timacdonald can we add these to `require-dev` instead please? Also, `nyholm/psr7` needs to be `^1.2` at minimum (see `suggest`).

Comment (timacdonald) [reply .github/workflows/tests.yml]: Done!

**Review by driesvints — APPROVED** [review]

**Pull request merged**